packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -523,26 +529,49 @@ export class SwapQuoter {
             gasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
         }
         // get the relevant orders for the makerAsset
-        let prunedOrders = await this._getSignedOrdersAsync(makerAssetData, takerAssetData);
+        const orderPromises: Array<Promise<SignedOrder[]>> = [
+            this._getSignedOrdersAsync(makerAssetData, takerAssetData),
+        ];
+        if (
+            options.rfqt &&
+            options.rfqt.intentOnFilling &&
+            options.apiKey &&
+            this._rfqtTakerApiKeyWhitelist.includes(options.apiKey)
+        ) {
+            if (!options.rfqt.takerAddress || options.rfqt.takerAddress === constants.NULL_ADDRESS) {
+                throw new Error('RFQ-T requests must specify a taker address');
+            }
+            orderPromises.push(
+                this._quoteRequestor.requestRfqtFirmQuotesAsync(
+                    makerAssetData,
+                    takerAssetData,
+                    assetFillAmount,
+                    marketOperation,
+                    options.apiKey,
+                    options.rfqt.takerAddress,
+                ),
+            );
+        }
+        const orders: SignedOrder[] = ([] as SignedOrder[]).concat(...(await Promise.all(orderPromises)));
```

Comment:
Would the following work & be more readable?

```
const resolvedPromises = await Promise.all(orderPromises);
const orders = _.flatten(resolvedPromises);
```

Reply:
What do you think of this? https://github.com/0xProject/0x-monorepo/pull/2541/commits/27ca75d94f2a874ef8c54b3ce7f50b34643b0d97

Reply:
As I said in the other comment, I generally would rather not introduce 3rd-party libs where it's not necessary.  And, I believe I have addressed the readability issue.  Please let me know what you think.

Reply:
New commit is looking better, still a little hard to read though.   

As mentioned tho -- we already have lodash as a dependency, and we can import just the function we care about (dont need to import whole library), i'd prefer just to use the lodash function here.  That being said, I'm ok as-is